### DISCUSSION TITLE
Overriding `DEFAULT_EXCLUDE` list

### DISCUSSION THREAD

**Discussion by rszamszur** [copier-org/copier#849] [Q&A] [answered]
I was wondering if there is a way to override `DEFAULT_EXCLUDE` list? Otherwise, if you are interested in such "feature" I could try to prepare a contribution for this.

https://github.com/copier-org/copier/blob/5ec47d95b95d26fee24a5e42d42cfefbe66c09ad/copier/template.py#L42-L51

Long story short:  I have a copier template that generates copier templates, and now `copier.yml` being by default excluded requires some workarounds. Currently, I just name `copier.yml` as `copier.ym.keep` and rename it in the task stage. It works, but there is a limiting factor of using `mv` coreutil to rename it. 

Core codebase: https://github.com/fastapi-mvc/fastapi-mvc
Generator template: https://github.com/fastapi-mvc/copier-generator

I think with copier being a superb tool for templating and scaffolding, it might be of interest for some people to use it for metaprogramming as well (in this case generating templates from templates).

Have a good one!

**Comment by yajo** [comment, marked as answer]
Hi there! Thanks for your initiative and good comments. I'm happy to help others with this project ❤️ 

I never designed it for metaprogramming, but indeed it should be perfectly possible. 

It seems to me like all you need to do is [define a valid exclude option](https://copier.readthedocs.io/en/latest/configuring/#exclude) in your main template.

When exclusions are defined in the template, the default is not used.

Would that work for you?

**Reply by rszamszur** [reply]
Yes, definitely.
